File: 2025/206. Reverse Linked List/Solution3.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode dummy = new ListNode(), cur = head, prev = dummy;
        dummy.next = head;
        while (cur != null && cur.next != null) {
            ListNode tmp = cur.next;
            cur.next = tmp.next;
            tmp.next = prev.next;
            prev.next = tmp;
        }
        return dummy.next;
    }
}

File: 2025/53. Maximum Subarray/solution3.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        cur, max_sum = 0, nums[0]
        for i in nums:
            cur += i
            if cur > max_sum:
                max_sum = cur
            if cur < 0:
                cur = 0
        return max_sum
